### PR TITLE
Improve scala.io.Source documentation

### PR DESCRIPTION
**Pull request by jbdeboer**
- fixes several typos
- clarifies non-obvious parameters
- adds code formatting and external class references

For methods with obvious parameters, I omitted `@param` documentation.  In all cases, I omitted the `@return` doc.

There are no code changes in this commit.

### CODE REVIEW

**Comment by lrytz** [comment]
Thanks for the PR, definitely an improvement!

File: src/library/scala/io/Source.scala [hunk 1]
```diff
@@ -26,66 +26,87 @@ object Source {
...
     val iter = iterable.iterator
   } withReset(() => fromIterable(iterable))
 
-  /** Creates a Source instance from a single character.
+  /** Creates a `Source` instance from a single character.
    */
   def fromChar(c: Char): Source = fromIterable(Array(c))
 
-  /** creates Source from array of characters, with empty description.
+  /** Creates `Source` from array of characters, with empty description.
```
Comment (lrytz) [inline src/library/scala/io/Source.scala:39]: There are missing articles, here and also many in later comments. Would you mind adding them? It should be something like "Creates **a** Source from **an** array of characters, with **an** empty description".

File: src/library/scala/io/Source.scala [hunk 2]
```diff
@@ -26,66 +26,87 @@ object Source {
...
+  /** Creates `Source` from a `String`, with no description.
    */
   def fromString(s: String): Source = fromIterable(s)
 
-  /** creates Source from file with given name, setting its description to
-   *  filename.
+  /** Creates `Source` from file with given name, setting its description to the
+   *  file's absolute path.
+   * 
+   *  @param name the pathname of the file. The format is dictated by the FileSystem.
```
Comment (lrytz) [inline src/library/scala/io/Source.scala:50]: Maybe add a reference to what is meant by `FileSystem`.

File: src/library/scala/io/Source.scala [hunk 3]
```diff
@@ -26,66 +26,87 @@ object Source {
...
    */
   def fromFile(uri: URI)(implicit codec: Codec): BufferedSource =
     fromFile(new JFile(uri))(codec)
 
-  /** creates Source from file with given file: URI
+  /** Creates `Source` from file with given `URI`, using the given encoding, setting
+   *  its description to the file's absolute path.
+   * 
+   *  @param uri the URI referencing the file.
+   *  @param enc the name of the `Charset` encoding to open the file with. e.g US-ASCII, UTF-16
```
Comment (lrytz) [inline src/library/scala/io/Source.scala:74]: some punctuation: `[..] to open this file with, e.g., "UTF-16."`

File: src/library/scala/io/Source.scala [hunk 4]
```diff
@@ -115,27 +140,38 @@ object Source {
   def fromRawBytes(bytes: Array[Byte]): Source =
     fromString(new String(bytes, Codec.ISO8859.name))
 
-  /** creates `Source` from file with given file: URI
+  /** Creates `Source` from file with given file: `URI`
+   *  Same as fromFile(uri)
```
Comment (lrytz) [inline src/library/scala/io/Source.scala:144]: The implementation is not the same as `fromFile(uri)`

File: src/library/scala/io/Source.scala [hunk 5]
```diff
@@ -115,27 +140,38 @@ object Source {
...
    */
   def fromURI(uri: URI)(implicit codec: Codec): BufferedSource =
     fromFile(new JFile(uri))(codec)
 
-  /** same as fromURL(new URL(s))(Codec(enc))
+  /** Creates `Source` from a given URL and encoding.
+   *  Same as fromURL(new URL(s))(Codec(enc))
```
Comment (lrytz) [inline src/library/scala/io/Source.scala:150]: The implementation is different..

**Comment by adriaanm** [comment]
Shall we accept the incremental improvement already, @lrytz or @gkossakowski? Ping @jbdeboer.

**Comment by gkossakowski** [comment]
My understanding is that some changes @lrytz commented on are misunderstandings that could mislead people reading Scaladocs. I'm deferring this until 2.11.4 with a hope that we'll hear back from @jbdeboer soon!

**Comment by lrytz** [comment]
Closing this, some of the changes are potentially confusing I would say.

**Pull request closed without merging**